### PR TITLE
feat: add d3m ticker

### PR DESCRIPTION
**Pull request by barrutko**

### CODE REVIEW

File: web3-functions/d3m-ticker/userArgs.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "threshold": "5000000000000000000000",
```
Comment (lucas-manuel) [inline web3-functions/d3m-ticker/userArgs.json]: Doesn't 5000 seem low?

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: web3-functions/d3m-ticker/userArgs.json (hunk 1 above)
Comment (barrutko) [reply web3-functions/d3m-ticker/userArgs.json]: Yeah, I agree it's pretty low, but frankly I wasn't setting this args as definitive for now. Future actions that are coming have even more args and I need to check with Sam before the deployment what are the values that he used. I'll set it to something more reasonable for now, like 10e6, but it'll need to be revisited before the very deployment anyway.

**Review by lucas-manuel — APPROVED** [review]

File: test/d3m-ticker.test.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                },
+                {
+                    target: addresses.mainnet.vat,
+                    callData: vat.interface.encodeFunctionData('urns', [ilk, addresses.mainnet.d3mPool]),
+                },
+            ])
+        ).returnData
+        const artAfter = vat.interface.decodeFunctionResult('urns', artAfterData).art
+
+        artDifferenceOnTheFork =
```
Comment (lucas-manuel) [inline test/d3m-ticker.test.ts]: Can we use a pinned block here and constants? This is basically the same logic as the code so would be better to have hard numbers and test the boundaries that way.

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: test/d3m-ticker.test.ts (hunk 2 above)
Comment (barrutko) [reply test/d3m-ticker.test.ts]: The whole setup is already pinned, so I just improved setup a bit and refactored this into a const.

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**